core/dbt/parser/manifest.py

```diff
@@ -125,6 +126,36 @@
 PERF_INFO_FILE_NAME = "perf_info.json"
 
 
+def extended_mashumaro_encoder(data):
+    return msgpack.packb(data, default=extended_msgpack_encoder, use_bin_type=True)
+
+
+def extended_msgpack_encoder(obj):
+    if type(obj) is datetime.date:
+        date_bytes = msgpack.ExtType(1, obj.isoformat().encode())
+        return date_bytes
```

Comment:
The [msgpack docs](https://github.com/msgpack/msgpack-python#packingunpacking-of-custom-data-type)  outline an alternative method to encoding custom types by returning something more like: 
```python
{'__datetime__': True, 'as_str': obj.strftime("%Y%m%dT%H:%M:%S.%f")}
```

It looks to be interchangeable with `ExtType` usage as you've implemented. The `ExtType` approach feels more readable + extensible to custom data types beyond datetime